src/main/java/com/project/security/JwtTokenProvider.java

```diff
@@ -0,0 +1,90 @@
+package com.project.security;
+
+import com.project.exception.InvalidJwtAuthenticationException;
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.Jws;
+import io.jsonwebtoken.JwtException;
+import io.jsonwebtoken.Jwts;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.stereotype.Component;
+
+import static io.jsonwebtoken.SignatureAlgorithm.*;
+
+import javax.annotation.PostConstruct;
+import javax.servlet.http.HttpServletRequest;
+import java.util.Base64;
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+
+@Component
+public class JwtTokenProvider {
+    @Value("${security.jwt.token.secret-key:secret}")
+    private String secretKey = "itacademy2020";
```

Comment:
You could move out this value to a property file so you won't need to expose it directly in the code.

Reply:
removed the "hard-coded" values